src/modules/ekf2/EKF2.cpp

```diff
@@ -2454,10 +2434,30 @@ void EKF2::UpdateGpsSample(ekf2_timestamps_s &ekf2_timestamps)
 
 		_ekf.setGpsData(gnss_sample);
 
+		const float geoid_height = altitude_ellipsoid - altitude_amsl;
+
+		if (_gps_time_usec == 0) {
+			_geoid_height_lpf.reset(geoid_height);
+
+		} else {
+			const float dt = 1e-6f * (_gps_time_usec - gnss_sample.time_us);
```

Comment:
```suggestion
			const float dt = 1e-6f * (gnss_sample.time_us - _gps_time_usec);
```

If you wanted to be paranoid you could also only setParameters() if  `gnss_sample.time_us > _gps_time_usec` to avoid potential non-sense if the primary GNSS instance changes.

Reply:
Ouch, nice catch. Fixed. I also added the paranoid check.